fluvio-protocol-core/src/encoder.rs

```diff
@@ -408,6 +408,35 @@ impl EncoderVarInt for Option<Vec<u8>> {
     }
 }
 
+impl EncoderVarInt for Vec<u8> {
```

Comment:
EncoderVarInt is not designed for vectors.  Is there reason for this?

Reply:
https://kafka.apache.org/protocol#protocol_types

Reply:
Oh, I understand now. I did this because it's implemented for [`Option<Vec<u8>>`](https://github.com/infinyon/fluvio-protocol/blob/0bf571ecf3b7ad09bcfb5ae89b0857e3b1788baf/fluvio-protocol-core/src/encoder.rs#L368-L409) to deal with https://github.com/simlay/fluvio/blob/7c99a0b96b69f54bbf6f702e0ff9c599b7dc8667/src/dataplane-protocol/src/record.rs#L139.